starwars_api/models.py

```diff
@@ -55,17 +64,47 @@ def __repr__(self):
 class BaseQuerySet(object):
 
     def __init__(self):
-        pass
+        self.page = 1
+        self.index = 0
+        self.json_data = api_client.get_people(**{'page': 1})
+
+        self.total_items = self.json_data['count']
+        self.results = self.json_data['results']
+        self.next_page = self.json_data['next'] # will be a URL until it is Null
+        
 
     def __iter__(self):
-        pass
+        return self 
 
     def __next__(self):
         """
         Must handle requests to next pages in SWAPI when objects in the current
         page were all consumed.
         """
-        pass
+        while self.results: # it should break when raise StopIteration
+            try:
+                current_page = self.results
+                individual_data = current_page[self.index]
+            except IndexError:
+                self.page += 1 # go to the next page
+                try:
+                    self.get_next_page(self.page)
+                except SWAPIClientError:
+                    raise StopIteration
+                self.index = 0 # you successfully got the next page so reset the index
+                current_page = self.results
+                individual_data = current_page[self.index]
+
+            self.index += 1
+            item = People(individual_data)
+
+            return item
+
+    def get_next_page(self, page_number):
+        json_data = api_client.get_people(**{'page': page_number})
+        self.results = json_data['results']
+        self.next = json_data['next']
```

Comment:
thank you. it means a lot to me cos I spent the majority of the time struggling with this cos there are so many variables to keep in mind. also I always try to make my code as self-explanatory as possible so yay, bonus. ^_^